hr_timesheet_activity_begin_end/models/account_analytic_line.py

```diff
@@ -15,9 +15,9 @@ class AccountAnalyticLine(models.Model):
     time_start = fields.Float(string="Begin Hour")
     time_stop = fields.Float(string="End Hour")
 
-    @api.one
     @api.constrains("time_start", "time_stop", "unit_amount")
     def _check_time_start_stop(self):
+        self.ensure_one()
```

Comment:
Here self could be a recordset, not necessarily a singleton.
This constraint should be rewritten this way:

```python
    @api.constrains("time_start", "time_stop", "unit_amount")
    def _check_time_start_stop(self):
        for line in self:
            value_to_html = self.env["ir.qweb.field.float_time"].value_to_html
            start = timedelta(hours=line.time_start)
            stop = timedelta(hours=line.time_stop)
            ....
            etc..
            ....
```

Reply:
You forgot the loop on self